constantdict/__init__.py

```diff
@@ -164,7 +176,12 @@ def delete(self, key: K) -> constantdict[K, V]:
 
     remove = delete
 
-    def update(self, other: dict[K, V]) -> constantdict[K, V]:  # type: ignore[override]
+    def update(self, other: Mapping[K, V]  # type: ignore[override]
+                      | SupportsKeysAndGetItem[K, V]
+                      | Iterable[tuple[K, V]]
+                      | Iterable[tuple[str, V]]
```

Comment:
It's my point of view that this doesn't need to imitate every bit of nonsense that the base `dict` allows. Also, not everything that succeeds at runtime must be allowed by the types.

Reply:
I understand. I'm not particular at all about the two `Iterable` type annotations, so I'll go ahead and remove them.